design-proposals/eim-standalone-managed-en.md

```diff
@@ -0,0 +1,236 @@
+- SENs being onboarded won't be equipped with desktop, keyboard or mouse. The only machine equipped with peripherals will be
+  the local developer/admin machine.
+- The design will re-use the current APIs of Onboarding Manager to drive IO/NIO-based onboarding.
+  We may just require small modifications to the Onboarding Manager and the gRPC interface.
+
+### Solution
+
+```mermaid
+sequenceDiagram
+%%{wrap}%%
+autonumber
+
+  box LightYellow Edge Node(s)
+  participant ctl as Standalone Onboarding CLI
+  end
+
+  participant user as User
+
+  box rgb(235,255,255) Edge Orchestrator
+
+  participant pa as Provisioning Nginx
+  participant om as Onboarding Manager
+  participant kc as KeyCloak
+  participant inv as Inventory / API
+
+  end
+
+  note over ctl,user: User installs Standalone EN, decides to scale out
+
+  note over pa,inv: OS profiles created beforehand, no Host/Instance pre-registration
+
+  user->>user: Retrieves standalone onboarding user credentials 
+  user->>ctl: Logs in to the EN and invokes CLI tool
+  
+  note over ctl: CLI triggered
+
+  ctl->>ctl: Checks local OS prerequisites
+
+  ctl->>ctl: Take user inputs and validate
+
+  ctl->>+pa: Get orchestrator CA certificate
+  pa-->>-ctl: [CA certificate]
+
+  ctl->>+kc: Retrieve JWT token for standalone onboarding user
+  kc->>-ctl: [JWT token]
+
+  ctl->>+om: Invoke OnboardStandaloneService gRPC (UUID, Serial Number, OS info, ...)
+
+  om->>om: Validate JWT standalone onboarding role
+
+  om->>inv: Query Inventory for OS resource by OS version and distribution
+
+  om->>om: Generate cloud-init for Standalone EN
+
+  om->>inv: Create Host and Instance resource, Set EN as Onboarded and Provisioned
+  om->>-ctl: Return cloud-init
+
+  ctl->>ctl: Save cloud-init & run manually or reboot
+
+  note over ctl: After OS reboot or manual cloud-init execution
+
+  ctl->>ctl: Activate Edge Node Agents
+```
+
+**Step 0.** A user already provisioned a set of Standalone ENs following the user guides and decides to scale out.
+They don't need to have direct access to the Edge Orchestrator UI/API (could be different personas on-site vs. remote administrator).
+Users don't need to perform any configuration on the Edge Orchestrator beforehand, but we assume that the Edge Orchestrator supports
+the OS version of the Standalone ENs. The remote administrator configures a special user/role for SEN onboarding.
+
+1. An on-site user retrieves user credentials for SEN onboarding from the remote administrator.
+2. An on-site user logs into the node and invokes the CLI tool that should already be installed on the EMT image.
+   The user enters into the interactive session with the CLI tool. 
+3. The CLI tool performs initial OS prerequisites checks - for instance, it can check if the UUID and SN are properly set on the OS. 
+4. A user is prompted for inputs. The user will be asked for the orchestrator FQDN, proxy settings (the already set proxy settings should be presented to the user),
```

Comment:
I was thinking... we need to provide a UX where customer can download a payload from the Orch instance (like what we download certs today). This payload can have certs, config and other artifacts for EMT-S node to basically connect to Orch (even OTP can be part of it). When customer copies this payload to all ENs (e.g. bulk scp) this agent on the EN (e.g. node agent) can start to communicate with Orch and move to normal operations.

Reply:
I really like this more than IO - that mimics in somehow the FDO flow